### ISSUE TITLE
Fish history autosuggest should be suppressed when true match is multiline

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#5597]
Fish's autosuggested ("ghost") history completions come from the list of single-line entries in the history file. In the event that the first matching history entry is multiline, that entry is not shown and instead fish displays matching history results from (possibly much) earlier on:

![image](https://user-images.githubusercontent.com/606923/51877460-e4c50e80-2331-11e9-8604-e705a9dcef84.png)

We either need to find a way to have the autocomplete suggest the multiline entry (e.g. just show the first line followed by &hellip; or "..and n additional lines" or something), show the multiline command as a suggestion with line breaks replaced with `;`, or else (my preference for now) simply suppress the autocomplete suggestion entirely as the behavior is inconsistent with the user's expectation that the suggestion will be the last history entry that started with the string in question.

As an example of a poor UX: a user is trained that to autocomplete the last history entry that matches what you want, enter a matching prefix then press → to complete it. Muscle memory and generally finding this to be the case means that the user may do so attempting to reproduce a multiline entry without reading the details, in which case they may execute the wrong command.

**Comment by faho** [comment]
> simply suppress the autocomplete suggestion entirely

I don't agree. It seems to me that it's better to offer _something_ than to offer nothing.

>a user is trained that to autocomplete the last history entry that matches what you want, enter a matching prefix then press → to complete it.

The suggestions are not just taken straight from history! We build synthetic commandlines that have never ever been run sometimes!

Whatever the suggestion, you still have to read it before you execute it.

**Comment by mqudsi** [comment]
> It seems to me that it's better to offer something than to offer nothing.

That's probably what led to the current behavior, and I very strongly disagree. Offering something inconsistent is *much* worse than offering nothing at all.

**Comment by floam** [comment]
I think I agree with @faho here - it's not meant to be the case that you can rely on the autosuggestion being the last thing in history. If anything, I believe we want to add more exceptions and more clever tricks to synthesize possibly useful command predictions which will defy your expectation here.

**Comment by ridiculousfish** [comment]
Yes autosuggestions are definitely not just a naive history search. As for multi-line commands, I think at one point fish showed the first line only, with ellipsis. But this wasn't really enough information to be useful (imagine just `begin...`). And showing the entire command would cause the command line to jump up, which is surprising. So I think the current behavior here is the right balance.

**Comment by floam** [comment]
I suppose we could show multi-line commands, with the newlines turned into `;`s?
